tests/admin/test_controller.py

```diff
@@ -2626,46 +2921,67 @@ def test_collections_post_errors(self):
         )
 
         with self.app.test_request_context("/", method="POST"):
+            self._db.begin_nested()
```

Comment:
I was thinking I could use savepoints to reduce the amount of duplicated setup code in these tests. However, it's not working the way I expected and things aren't getting rolled back at the end of the test. I will probably revert all of the changes related to this and repeatedly create new admins instead, but I'd like to know if you have other ideas.